ptxcompiler/patch.py

```diff
@@ -183,7 +206,24 @@ def patch_needed():
     logger.debug("CUDA Driver version %s.%s" % driver_version)
     logger.debug("CUDA Runtime version %s.%s" % runtime_version)
 
-    return driver_version < runtime_version
+    return driver_version, runtime_version
+
+
+def safe_get_versions():
+    if check_disabled_in_env():
+        try:
+            # allow user to specify driver/runtime versions manually, if necessary
+            driver_version = os.environ["PTXCOMPILER_KNOWN_DRIVER_VERSION"].split(".")
+            runtime_version = os.environ["PTXCOMPILER_KNOWN_RUNTIME_VERSION"].split(".")
```

Comment:
Won't they be of the form `x.y`, e.g. `11.7`?

Reply:
So the driver call `cuDriverGetVersion` returns just an integer (e.g. 11070), but I see that `ptxcompiler`'s patching turns that into a pair of integers `(11, 70)`. So the question is whether the user should specify:

- What the C API would return
- What the ptxcompiler call returns
- Something else (as here)

Reply:
This might just be personal preference but I tend to specify things in `x.y` form. I've never referred to a cuda version in the `cuDriverGetVersion` form outside of dealing with those specific APIs. I also think `11.5` is easier to reason about than having to put the string `(11, 5)` in an environment variable - it feels like it leaks some implementation detail that we work with these versions as python tuples. 

If I'm looking up something about a cuda version, I'm generally searching for results that match the `x.y` pattern on the web.

Reply:
Sure, we just need to make sure whatever the user specifies it gets turned into something appropriate for later comparison.

Reply:
Oh I see. We can validate that the result here is a 2-tuple and that its elements are intable, how does that sound?

Reply:
What does `get_versions` return? We should allow the user to specify something that we can munge into the equivalent, I think.

Reply:
Do you mean something like
```suggestion
            driver_version = os.environ["PTXCOMPILER_KNOWN_DRIVER_VERSION"].split(".")
            runtime_version = os.environ["PTXCOMPILER_KNOWN_RUNTIME_VERSION"].split(".")
            driver_version, runtime_version = (
                tuple(map(int, driver_version)), 
                tuple(map(int, runtime_version)
            )
```

and then catching on a `ValueError` as well below?

Reply:
added at https://github.com/rapidsai/ptxcompiler/pull/28/commits/1b8d703955b12a5764fb13b666c0e5818b64fee1